routers/routes/routes.go

```diff
@@ -591,6 +591,7 @@ func RegisterRoutes(m *macaron.Macaron) {
 		m.Group("", func() {
 			m.Get("/^:type(issues|pulls)$", repo.RetrieveLabels, repo.Issues)
 			m.Get("/^:type(issues|pulls)$/:index", repo.ViewIssue)
+			m.Get("/pulls/:index.diff", repo.DownloadPullDiff)
```

Comment:
There should be check that user has rights to access `models.UnitTypePullRequests`

Reply:
why isn't it done for the route before ? How's \.diff different from the non-diff URL ?

Reply:
If you would move it to this part of route than you would not need to check rights as all group has right check already:
https://github.com/strk/gitea/blob/43c2bfa2d016d899936a178b5b5676e27d13c8dc/routers/routes/routes.go#L629

Reply:
There is no check for group as these routes are common for issues and PR and each have it's own rights

Reply:
moved with c60fa34b